hip/components/memory.hip.hpp

```diff
@@ -60,6 +60,12 @@ struct gcc_atomic_intrinsic_type_map<double> {
 };
 
 
+template <>
+struct gcc_atomic_intrinsic_type_map<__half> {
```

Comment:
maybe leave this undefined, so we will get compiler errors, when we try to use hip fp16 atomics.